subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -926,15 +957,37 @@ int ull_scan_aux_stop(struct ll_scan_aux_set *aux)
 
 	/* Abort LLL event if ULL scheduling not used or already in prepare */
 	if (err == -EALREADY) {
-		ret = ull_disable(&aux->lll);
-		if (ret) {
-			return -EBUSY;
+		err = ull_disable(&aux->lll);
+		if (err) {
+			return err;
 		}
 
 		mfy.fp = flush;
+
+	} else if (!IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC)) {
+		/* ULL scan auxiliary PDU reception scheduling stopped
```

Comment:
Just comment.

In my humble opinion, there should be a documentation that puts this kind of functions in all possible flows (e.g.  some diagrams) to make it easier to understand. This kind of functions depend on some global state and it is hard to put it into right context to get thorough understanding. Comments doesn't solve the problem. Comments are helpful for a person that is deep into the context.

Reply:
I agree with this, but this is outside the scope of this PR. @cvinayak I believe already has plans to document the LL once the crunch with Audio is complete.